syntax = "proto3";

package gossip;

service GossipService {
  rpc SendMessage(GossipMessage) returns (MessageResponse);
}

message GossipMessage {
  string message = 1;
  string sender_id = 2;
  int64 timestamp = 3;
  double latency_ms = 4;
}

message MessageResponse {
  string message = 1;
  bool received = 2;
}